kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/dns/LinuxDns.java

```diff
@@ -367,6 +370,17 @@ private synchronized String getRealPath(File file) throws KuraException {
 
     private synchronized void writeDnsFile(Set<IPAddress> servers) {
         logger.debug("Writing DNS servers to file");
+        // Check if DNS_FILE_NAME is a symlink and it is broken or points to unmanaged location.
+        // In this case, delete it.
+        try {
```

Comment:
Maybe move this block to a specific method with a meaningful name, in order to be able to remove the comment.